src/form-elements/header/edit-donation.ts

```diff
@@ -0,0 +1,402 @@
+
+  @property({ type: Object }) private error?: TemplateResult;
+
+  @query('#custom-amount-button') customAmountButton?: HTMLInputElement;
+
+  @query('#custom-amount-input') customAmountInput?: HTMLInputElement;
+
+  private currencyValidator: CurrencyValidator = new CurrencyValidator();
+
+  render(): TemplateResult {
+    return html`
+      <form-section number="1" headline="Choose a frequency">
+        <ul class="frequency-selector">
+          ${this.frequencyTemplate}
+        </ul>
+      </form-section>
+
+      <form-section number="2" headline="Choose an amount (USD)">
+        <ul class="amount-selector">
+          ${this.presetAmountsTemplate}
+          <li class="custom-amount">${this.customAmountTemplate}</li>
+        </ul>
+
+        <div class="errors">
+          ${this.error}
+        </div>
+
+        <div class="cover-fees-container">
+          <input type="checkbox" id="cover-fees" @input=${this.coverFeesChecked} />
+          <label for="cover-fees">
+            I'll generously add ${currency(this.donationInfo.fee, { symbol: '$' }).format()} to
+            cover the transaction fees so you can keep 100% of my donation.
+          </label>
+        </div>
+      </form-section>
+    `;
+  }
+
+  private get frequencyTemplate(): TemplateResult {
+    return html`
+      <li>
+        ${this.getRadioButton({
+          group: SelectionGroup.DonationType,
+          value: DonationType.OneTime,
+          displayText: 'One time',
+          checked: this.donationInfo.donationType === DonationType.OneTime,
+        })}
+      </li>
+
+      <li>
+        ${this.getRadioButton({
+          group: SelectionGroup.DonationType,
+          value: DonationType.Monthly,
+          displayText: 'Monthly',
+          checked: this.donationInfo.donationType === DonationType.Monthly,
+        })}
+      </li>
+    `;
+  }
+
+  private get presetAmountsTemplate(): TemplateResult {
+    return html`
+      ${this.amountOptions.map(amount => {
+        const checked = !this.customAmountButton?.checked && amount === this.donationInfo.amount;
+        return html`
+          <li>
+            ${this.getRadioButton({
+              group: SelectionGroup.Amount,
+              value: `${amount}`,
+              displayText: `$${amount}`,
+              checked: checked,
+            })}
+          </li>
+        `;
+      })}
+    `;
+  }
+
+  private getRadioButton(options: {
+    group: SelectionGroup;
+    value: string;
+    displayText: string;
+    checked: boolean;
+  }): TemplateResult {
+    const radioId = `${options.group}-${options.value}-option`;
+    return html`
+      <div class="selection-button">
+        <input
+          type="radio"
+          name=${options.group}
+          value=${options.value}
+          id=${radioId}
+          .checked=${options.checked}
+          @change=${this.radioSelected}
+        />
+        <label for=${radioId}>${options.displayText}</label>
```

Comment:
suggestion: wrap input in label attribute so we don't need the for/id attrs
the label will still be clickable

![image](https://user-images.githubusercontent.com/7840857/88889720-66cb1780-d1f5-11ea-8e06-1d2f89c36d4a.png)